Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,134 @@
+            _pathIndex.Add("C#","/Audio/Effects/RingtoneNotes/c#.ogg");
+            _pathIndex.Add("D","/Audio/Effects/RingtoneNotes/d.ogg");
+            _pathIndex.Add("D#","/Audio/Effects/RingtoneNotes/d#.ogg");
+            _pathIndex.Add("E","/Audio/Effects/RingtoneNotes/e.ogg");
+            _pathIndex.Add("F","/Audio/Effects/RingtoneNotes/f.ogg");
+            _pathIndex.Add("F#","/Audio/Effects/RingtoneNotes/f#.ogg");
+            _pathIndex.Add("G","/Audio/Effects/RingtoneNotes/g.ogg");
+            _pathIndex.Add("G#","/Audio/Effects/RingtoneNotes/g#.ogg");
+        }
+
+        //Event Functions
+
+        private void RingerPlayRingtone (EntityUid uid, RingerComponent ringer, RingerPlayRingtoneMessage args) {
+            ringer.isPlaying = true; //Allows the overridden Update function to play the ringtone based off of the ringer component's stored ringtonePathArr
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void UpdateRingerUserInterfaceDriver(EntityUid uid, RingerComponent ringer, RingerRequestUpdateInterfaceMessage args) { //Seprate Drivers are needed because UpdateRingerUserInterface() and UpdateRingerComponent() are called by other functions aside from the events themselves
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void RingerUpdateRingtoneDriver(EntityUid uid, RingerComponent ringer, RingerSetRingtoneMessage args) {
+            UpdateRingerComponent(ringer, args.ringtoneStringArgArr);
+        }
+
+        public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, ComponentInit args) { //Randomizes the ringtone
+            string[] possibleNotes = new string[12]{"A","B","C","D","E","F","G","A#","C#","D#","F#","G#"};
+            string[] randomizedRingerStringArr = new string[4];
+            var randomNote = new Random();
+
+            for(int count = 0; count < 4; count++) {
+                int randomNoteIndex = randomNote.Next(possibleNotes.Length);
+                randomizedRingerStringArr[count] = possibleNotes[randomNoteIndex];
+            }
+
+            UpdateRingerComponent(ringer, randomizedRingerStringArr);
+
+        }
+
+        //Non Event Functions
+
+        private bool UpdateRingerComponent(RingerComponent ringer, string[] ringerStringArr) {
+            bool _isNull = false;
+            string[] _ringtonePathArr = new string[4];
+            for(int strngCount = 0; strngCount < 4; strngCount++) { //Prevents "null" (Not technically null but "" values cant be passed) values from being passed to the component
+                if(ringerStringArr[strngCount].Equals("")) {
+                    _isNull = true;
+                }
+            }
+            if(_isNull){return false;};
+            ringer.ringtoneStringArr = ringerStringArr;
+
+            for(int strngCount = 0; strngCount < 4; strngCount++) {
+                _ringtonePathArr[strngCount] = _pathIndex[ringerStringArr[strngCount]];
+            }
```

Comment:
Some of this is kind of unnecessarily hard to read, also just add the 1 character and call it string.

You can also just use string.IsNullOrEmpty.